lib/Ast.ml

```diff
@@ -2332,22 +2345,9 @@ end = struct
         when e0 == exp ->
           false
       | Pexp_record (_, Some e0) when e0 == exp -> true
-      | Pexp_sequence
-          ( ( { pexp_desc=
-                  Pexp_extension
-                    ( _
-                    , PStr
-                        [ { pstr_desc=
-                              Pstr_eval ({pexp_desc= Pexp_sequence _; _}, [])
-                          ; _ } ] )
-              ; _ } as lhs )
-          , _ )
-        when lhs == exp ->
-          true
-      | Pexp_sequence (lhs, _) when lhs == exp ->
-          exposed_right_exp Let_match exp
-      | Pexp_sequence (_, rhs) when rhs == exp -> false
-      | _ -> parenze () )
+      | Pexp_sequence (lhs, rhs) -> exp_in_sequence lhs rhs exp
+      | _ -> Exp.has_trailing_attributes exp || parenze () )
+    | _, exp when Exp.has_trailing_attributes exp -> true
```

Comment:
I didn't expect that this case can be moved. I found no diffs with `test_branch`.
This simplifies the code quite a bit.